src/snowflake/cli/_plugins/snowpark/snowpark_entity.py

```diff
@@ -1,29 +1,297 @@
+        Returns:
+        """
+        # 0 Create a directory for the entity
+        if not output_dir:
+            output_dir = self.root / "output" / self.model.stage
+        output_dir.mkdir(parents=True, exist_ok=True)  # type: ignore
+
+        output_files = []
+
+        # 1 Check if requirements exits
+        if (self.root / "requirements.txt").exists():
+            download_results = self._process_requirements(
+                bundle_dir=output_dir,  # type: ignore
+                archive_name="dependencies.zip",
+                requirements_file=SecurePath(self.root / "requirements.txt"),
+                ignore_anaconda=ignore_anaconda,
+                skip_version_check=skip_version_check,
+                index_url=index_url,
+                allow_shared_libraries=allow_shared_libraries,
+            )
+
+        # 3 get the artifacts list
+        artifacts = self.model.artifacts
+
+        for artifact in artifacts:
+            output_file = output_dir / artifact.dest / artifact.src.name
+
+            if artifact.src.is_file():
+                output_file.mkdir(parents=True, exist_ok=True)
+                SecurePath(artifact.src).copy(output_file)
+            elif artifact.is_dir():
+                output_file.mkdir(parents=True, exist_ok=True)
+
+            output_files.append(output_file)
+
+        return output_files
+
+    def check_if_exists(
+        self, action_ctx: ActionContext
+    ) -> bool:  # TODO it should return current state, so we know if update is necessary
+        try:
+            current_state = self.action_describe(action_ctx)
+            return True
+        except ProgrammingError:
+            return False
+
+    def get_deploy_sql(self, mode: DeployMode):
+        query = [
+            f"{mode.value} {self.model.type.upper()} {self.identifier}",
+            "COPY GRANTS",
+            f"RETURNS {self.model.returns}",
+            f"LANGUAGE PYTHON",
+            f"RUNTIME_VERSION '{self.model.runtime or DEFAULT_RUNTIME}'",
+            f"IMPORTS={','.join(self.model.imports)}",  # TODO: Add source files here after introducing bundlemap
+            f"HANDLER='{self.model.handler}'",
+        ]
+
+        if self.model.external_access_integrations:
+            query.append(self.model.get_external_access_integrations_sql())
+
+        if self.model.secrets:
+            query.append(self.model.get_secrets_sql())
+
+        if self.model.type == "procedure" and self.model.execute_as_caller:
+            query.append("EXECUTE AS CALLER")
+
+        return "\n".join(query)
+
+    def get_describe_sql(self):
+        return f"DESCRIBE {self.model.type.upper()} {self.identifier}"
+
+    def get_drop_sql(self):
+        return f"DROP {self.model.type.upper()} {self.identifier}"
+
+    def get_execute_sql(self, execution_arguments: List[str] | None = None):
+        raise NotImplementedError
+
+    def get_usage_grant_sql(self, app_role: str):
```

Comment:
```suggestion
    def get_usage_grant_sql(self, role: str):
```
There's nothing app specific in this query as far as I can tell

Reply:
Done